File: leetcode/best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
/**************************************
* @author dujiong
* @date 2016.12.7
* @version V0.1
**************************************/

class Solution {
public:
    int maxProfit(vector<int> &prices) {
        if(prices.size()==0 || prices.size()==1){
            return 0;
        }
        int maxprofit = 0;
        int min = prices[0];
        for(int i=1;i<prices.size();i++){
            int temp = prices[i]-min;
            if(temp>maxprofit){
                maxprofit=temp;
            }
            if(prices[i]<min){
                min=prices[i];
            }
        }
        return maxprofit;
    }
};

File: leetcode-golang/Best Time to Buy and Sell Stock/Best Time to Buy and Sell Stock.go
func maxProfit(prices []int) int {
	if len(prices) <= 1 {
		return 0
	}
	result := 0
	local := prices[0]
	for i := 1; i < len(prices); i++ {
		if prices[i]-local > result {
			result = prices[i] - local
		}
		if prices[i] < local {
			local = prices[i]
		}
	}
	return result
}